src/shork.gleam

```diff
@@ -146,13 +185,24 @@ pub fn query(sql: String) -> Query(Nil) {
   Query(sql:, parameters: [], row_decoder: decode.success(Nil))
 }
 
+pub type TransactionError {
+  TransactionQueryError(QueryError)
+  TransactionRolledBack(String)
+}
+
 @external(erlang, "shork_ffi", "query")
 pub fn run_query(
   a: Connection,
   b: String,
   c: List(Value),
 ) -> Result(#(List(String), List(dynamic.Dynamic)), QueryError)
 
+@external(erlang, "shork_ffi", "transaction")
+pub fn transaction(
+  connection: Connection,
+  cb: fn(Connection) -> Result(t, String),
```

Comment:
I'd probably use `callback` or something as the argument name, in case somebody wants to use named arguments.

Reply:
Also: Is that String error argument the same as pog? I think, a generic type might be better.

Reply:
It's the same as in pog. But I added a generic type.